src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -979,12 +973,14 @@ int MacroAssembler::push_reg(unsigned int bitset, Register stack) {
 
   unsigned char regs[32];
   int count = bitset_to_regs(bitset, regs);
+  // reserve one slot to align for odd count
+  int offset = is_even(count) ? 0 : wordSize;
 
   if (count) {
-    addi(stack, stack, - count * wordSize);
+    addi(stack, stack, - count * wordSize - offset);
   }
   for (int i = count - 1; i >= 0; i--) {
-    sd(as_Register(regs[i]), Address(stack, (count -1 - i) * wordSize));
+    sd(as_Register(regs[i]), Address(stack, (count -1 - i) * wordSize + offset));
```

Comment:
whitespace is missing here. Should be: (count - 1 - i)

Reply:
Fixed.